verification/VerifyTasks.wdl

```diff
@@ -331,42 +331,107 @@ task CompareLooms {
   }
 }
 
-task CompareH5adFiles {
+task CompareH5adFilesATAC {
   input {
     File truth_h5ad
     File test_h5ad
-
-    Int cpu = 1
-    String docker = "gcr.io/gcp-runtimes/ubuntu_16_0_4:latest"
-    Int disk_size_gb = ceil(size(truth_h5ad, "GiB") + size(test_h5ad, "GiB")) + 20
-    Int memory_mb = ceil(size(truth_h5ad, "MiB") + size(test_h5ad, "MiB") * 2)
+    String docker = "python:3.10.0-buster"
+    Int disk_size_gb = ceil(size(truth_h5ad, "GiB") + size(test_h5ad, "GiB")) + 200
+    Int memory_gb = 20
   }
 
   command <<<
 
     set -eo pipefail
 
-    # calculate file sizes
+    pip3 install anndata
+    
+    python3 <<CODE
+    
+    import anndata as ad
+    import numpy as np
+    import pandas as pd
+    
+    truth_h5ad = "~{truth_h5ad}"
+    test_h5ad = "~{test_h5ad}"
+    truth = ad.read_h5ad(truth_h5ad)
+    test = ad.read_h5ad(test_h5ad)
+    
+    truth_obs = pd.DataFrame(truth.obs)
+    test_obs = pd.DataFrame(test.obs)
+    
+    print("Now running obs equivalence check")
+    
+    if truth_obs.equals(test_obs)==True:
+        print("true")
+    else:
+        exit("Files are not identical")
+    
+    print("Done running matrix equivalence check")
+    
+    CODE 
+  >>>
 
-    TRUTHSIZE=$(stat -c%s ~{truth_h5ad})
-    TESTSIZE=$(stat -c%s ~{test_h5ad})
+  runtime {
+    docker: docker
+    disks: "local-disk ${disk_size_gb} HDD"
+    memory: "${memory_gb} GiB"
+    preemptible: 3
+  }
+}
 
-    if [ "$TRUTHSIZE" == "$TESTSIZE" ]; then
-      echo "H5ad file sizes are identical"
-    else
-      echo "H5ad file sizes are NOT identical"
-      exit 1
-    fi
+task CompareH5adFilesGEX {
+  input {
+    File truth_h5ad
+    File test_h5ad
+    String docker = "python:3.10.0-buster"
+    Int disk_size_gb = ceil(size(truth_h5ad, "GiB") + size(test_h5ad, "GiB")) + 200
+    Int memory_mb = ceil(size(truth_h5ad, "MiB") + size(test_h5ad, "MiB") * 5)
```

Comment:
We should fix memory

Reply:
@khajoue2 Thanks for taking a look. For this file comparison, the dynamic memory worked. What do you suggest changing?